core/src/main/java/hudson/node_monitors/InodesMonitor.java

```diff
@@ -0,0 +1,162 @@
+ *
+ * Uses the <code>df</code> system command.
+ */
+public class InodesMonitor extends NodeMonitor {
+
+	private static final Logger LOGGER = Logger.getLogger(InodesMonitor.class.getSimpleName());
+
+	private static final String DEFAULT_OFFLINE_THRESHOLD = "95%";
+
+	public final String inodesPercentThreshold;
+
+	/**
+	 * @param inodesPercentThreshold
+	 *            threshod expected to be a percentage between 0% and 99% (% required). "5%" is correct. "5" is not.
+	 *
+	 * @throws ParseException
+	 *             if unable to parse.
+	 */
+	@DataBoundConstructor
+	public InodesMonitor(String inodesPercentThreshold) throws ParseException {
+		if (inodesPercentThreshold == null) {
+			inodesPercentThreshold = DEFAULT_OFFLINE_THRESHOLD;
+		}
+		parse(inodesPercentThreshold); // checks it parses
+		this.inodesPercentThreshold = inodesPercentThreshold;
+	}
+
+	public InodesMonitor() {
+		inodesPercentThreshold = DEFAULT_OFFLINE_THRESHOLD;
+	}
+
+	@VisibleForTesting
+	static int parse(String threshold) throws ParseException {
+		if (!threshold.matches("\\d?\\d%")) {
+			throw new ParseException(threshold, 0);
+		}
+		return Integer.parseInt(threshold.substring(0, threshold.length() - 1));
+	}
+
+	@Override
+	public Object data(Computer computer) {
+		String currentValueStr = (String) super.data(computer);
+		if (currentValueStr == null || currentValueStr.contains(Messages.InodesMonitor_NotApplicable())) {
+			return currentValueStr;
+		}
+		try {
+			int currentValue = parse(currentValueStr);
+			String currentState = "current=" + currentValue + ",threshold=" + inodesPercentThreshold;
+			String computerName = computer.getName();
+			// master has no nodeName
+			if ("".equals(computer.getName())) {
+				computerName = hudson.model.Messages.Hudson_Computer_DisplayName();
+			}
+
+			if (currentValue >= parse(inodesPercentThreshold)) {
+				OfflineCause offlineCause = OfflineCause.create(Messages._InodesMonitor_MarkedOffline(computerName, currentState));
+				if (((InodesUseInPercentMonitorDescriptor) getDescriptor()).markOffline(computer, offlineCause)) {
+					String inodesmonitor_markedOffline = Messages.InodesMonitor_MarkedOffline(computerName, currentState);
+					LOGGER.warning(inodesmonitor_markedOffline);
+				}
+			}
+			else {
+				if (((InodesUseInPercentMonitorDescriptor) getDescriptor()).markOnline(computer)) {
+					LOGGER.warning(Messages.InodesMonitor_MarkedOnline(computerName, currentState));
+				}
+			}
+		}
+		catch (ParseException e) {
+			// Shouldn't happen since received value is the one already provided by internal GetInodesUseInPercent
+			throw new IllegalStateException("WTF? Can't parse " + currentValueStr + " as integer percentage", e);
+		}
+		return currentValueStr;
+	}
+
+	@Override
+	public final String getColumnCaption() {
+		// Hide to non-admins
+		return Jenkins.getInstance().hasPermission(Jenkins.ADMINISTER) ? super.getColumnCaption() : null;
+	}
+
+	@Extension
```

Comment:
Was unaware of that. Used the existing NodeMonitor impls to get mine running but they are certainly too old to have got that applied. Makes sense. Will do.